client/src/core/RuntimeTypes.res

```diff
@@ -0,0 +1,189 @@
+// The types that are used for execution. For all type definitions, see RuntimeTypes.fs
+
+module Belt = {
+  include (Belt: module type of Belt with module Map := Belt.Map and module Result := Belt.Result)
+
+  module Result = {
+    include Belt.Result
+
+    let pp = (
+      okValueFormatter: (Format.formatter, 'okValue) => unit,
+      errValueFormatter: (Format.formatter, 'errValue) => unit,
+      fmt: Format.formatter,
+      value: t<'okValue, 'errValue>,
+    ) => {
+      switch value {
+      | Ok(value) =>
+        Format.pp_print_string(fmt, "Ok")
+        okValueFormatter(fmt, value)
+      | Error(value) =>
+        Format.pp_print_string(fmt, "Error")
+        errValueFormatter(fmt, value)
+      }
+    }
+  }
+
+  module Map = {
+    include (Belt.Map: module type of Belt.Map with module String := Belt.Map.String)
+
+    module String = {
+      include Belt.Map.String
+
+      let pp = (
+        valueFormatter: (Format.formatter, 'value) => unit,
+        fmt: Format.formatter,
+        map: t<'value>,
+      ) => {
+        Format.pp_print_string(fmt, "{ ")
+        Belt.Map.String.forEach(map, (key, value) => {
+          Format.pp_print_string(fmt, key)
+          Format.pp_print_string(fmt, ": ")
+          valueFormatter(fmt, value)
+          Format.pp_print_string(fmt, ",  ")
+        })
+        Format.pp_print_string(fmt, "}")
+        ()
+      }
+    }
+  }
+}
+
+@ppx.deriving(show) type rec id = ID.t
+@ppx.deriving(show) type rec tlid = TLID.t
+
+module FQFnName = {
+  @ppx.deriving(show({with_path: false}))
+  type rec stdlibFnName = {module_: string, function_: string, version: int}
+
+  @ppx.deriving(show({with_path: false})) type rec userFnName = string
+
+  @ppx.deriving(show({with_path: false}))
+  type rec packageFnName = {
+    owner: string,
+    package: string,
+    module_: string,
+    function_: string,
+    version: int,
+  }
+
+  @ppx.deriving(show({with_path: false}))
+  type rec t =
+    | User(userFnName)
+    | Stdlib(stdlibFnName)
+    | Package(packageFnName)
+}
+
+@ppx.deriving(show({with_path: false}))
+type rec isInPipe =
+  | InPipe(id)
+  | NotInPipe
+
+@ppx.deriving(show({with_path: false}))
+type rec pattern =
+  | PVariable(id, string)
+  | PConstructor(id, string, list<pattern>)
+  // TODO: support char
+  | PInteger(id, int64)
+  | PBool(id, bool)
+  | PString(id, string)
+  | PFloat(id, float)
+  | PNull(id)
+  | PBlank(id)
+
+and sendToRail =
+  | Rail
+  | NoRail
+
+and expr =
```

Comment:
I'm not seeing Tuple types in this file for some reason

Reply:
Yes it should. Wrote this before they merged, nice catch.